trillian/ctfe/cert_checker.go

```diff
@@ -93,6 +93,20 @@ func ValidateChain(rawChain [][]byte, validationOpts CertValidationOpts) ([]*x50
 		return nil, errors.New("rejecting unexpired certificate")
 	}
 
+	// Check for unwanted extension types, if required.
+	if len(validationOpts.rejectExtIds) != 0 {
+		badIDs := make(map[string]bool)
```

Comment:
Could build this map once at startup and attach it to the instance opts maybe?

Reply:
I thought about that, but I left it for now because it felt like the API should be "here's a list of things to ignore" and forcing the caller to pre-process that list into a `map` felt wrong, we could have a c'tor to do it (and in fact there is a c'tor, but it's already unwieldy with too many args), having some optional unexported map in the opts was another thought, but then that'd require a sync.Once/mutex/something in there too which seemed a bit heavy.

Ideally we'd refactor the c'tor to be a builder (`new Opts().WithExpiryDate(x).With...With.. etc. and pop it in there, but that's a bigger change that probably deserves its own PR.

I'll pop a TODO next to this.

Reply:
It just seemed like it could be handled more similarly to the EKU stuff. Was just a drive-by.

The API is more Set<thing> rather than List<thing> so map[thing]bool seems like it could be appropriate in future.

Reply:
Actually it was the `ExtKeyUsages` which helped me come to this since it's also specified as a slice.
I appreciate all the drive-by comments too - thank you for giving them! :)

Reply:
OK in practice I think we can assume that EKU and OID exclusions won't be a long list so I think this fine to go on with.